nodejs/week4/spam-detector/index.js

```diff
@@ -0,0 +1,70 @@
+class Email {
+  constructor(subject, body) {
+    this.subject = subject;
+    this.body = body;
+  }
+}
+
+const spamWords = ["viagra", "offer", "free", "business proposal"];
+
+class SpamDetector {
+  isSpam(email) {
+    const subject = email.subject;
+    const body = email.body;
+    const string = subject + body;
+
+    //more than 60% uppercase characters
+    let uppercaseCharacters = 0;
+    let uppercaseCharactersArray = [];
+    let numberOfCharacters = subject.length;
+    const subjectWithoutSpaces = subject.split(" ").join();

```

Comment:
`subjectWithoutSpaces` contains all the words joined with commas. To substitute all of the spaces by empty strings, you have to pass an empty string to `join`. Also, keep in mind, that after you change it, `numberOfCharacters` will not always be equal `subjectWithoutSpaces.length`, as you will remove spaces. To fix it, you could move declaration `numberOfCharacters` on the line below, and make it equal to `subjectWithoutSpaces.length`

Reply:
yes makes perfect sense, changed